cmd/process-agent/collector.go

```diff
@@ -69,14 +69,29 @@ func NewCollector(cfg *config.AgentConfig) (Collector, error) {
 		return Collector{}, err
 	}
 
+	var processCheck *checks.ProcessCheck
+	var processRTCheck *checks.RTProcessCheck
+
 	enabledChecks := make([]checks.Check, 0)
 	for _, c := range checks.All {
 		if cfg.CheckIsEnabled(c.Name()) {
 			c.Init(cfg, sysInfo)
 			enabledChecks = append(enabledChecks, c)
+
+			if check, ok := c.(*checks.ProcessCheck); ok {
+				processCheck = check
+			}
+			if check, ok := c.(*checks.RTProcessCheck); ok {
+				processRTCheck = check
+			}
```

Comment:
https://stackoverflow.com/questions/16725921/go-switch-vs-if-else-efficiency

It seems like switch would be equivalent to if-else. I guess there was a slight inefficiency here in that if it passed the first if statement, it would try the second one anyway.